connector/subgraph/src/Agreement.ts

```diff
@@ -0,0 +1,21 @@
+import { Proposal as ProposalEntity } from '../generated/schema'
+import {
+  Agreement as AgreementContract,
+  ActionDisputed as ActionDisputedEvent,
+} from '../generated/templates/Agreement/Agreement'
+import { getProposalEntityId } from './helpers/index'
+import { STATUS_DISPUTED } from './statuses'
+
+/* eslint-disable @typescript-eslint/no-use-before-define */
+
+export function handleActionDisputed(event: ActionDisputedEvent): void {
+  const agreementApp = AgreementContract.bind(event.address)
+  const actionData = agreementApp.getAction(event.params.actionId)
+  const challengeData = agreementApp.getChallenge(event.params.challengeId)
+  const proposalId = getProposalEntityId(actionData.value0, actionData.value1)
+
+  const proposal = ProposalEntity.load(proposalId)!
```

Comment:
Let's use `getProposalEntity`
```suggestion
  const proposal = getProposalEntity(actionData.value0, actionData.value1)
```